src/lib/crud.service.ts

```diff
@@ -86,79 +90,76 @@ export class CrudService<T extends BaseEntity> {
+            return this.repository.save(_.assign(entity, crudUpdateOneRequest.body));
+        });
     };
 
     readonly reservedDelete = async (crudDeleteOneRequest: CrudDeleteOneRequest<T>): Promise<T> => {
         if (this.primaryKey.length === 0) {
             throw new ConflictException('cannot found primary key from entity');
         }
+        return this.findOne(crudDeleteOneRequest.params as unknown as FindOptionsWhere<T>, false).then(async (entity: T | null) => {
+            if (!entity) {
+                throw new NotFoundException();
+            }
 
-        return this.repository
-            .findOne({
-                where: crudDeleteOneRequest.params as unknown as FindOptionsWhere<T>,
-            })
-            .then(async (entity: T | null) => {
-                if (!entity) {
-                    throw new NotFoundException();
-                }
-
-                if (crudDeleteOneRequest.author) {
-                    _.merge(entity, { [crudDeleteOneRequest.author.property]: crudDeleteOneRequest.author.value });
-                    await this.repository.save(entity);
-                }
+            if (crudDeleteOneRequest.author) {
+                _.merge(entity, { [crudDeleteOneRequest.author.property]: crudDeleteOneRequest.author.value });
+                await this.repository.save(entity);
+            }
 
-                await (crudDeleteOneRequest.softDeleted ? entity.softRemove() : entity.remove());
-                return entity;
-            });
+            await (crudDeleteOneRequest.softDeleted ? entity.softRemove() : entity.remove());
+            return entity;
+        });
     };
 
     readonly reservedRecover = async (crudRecoverRequest: CrudRecoverRequest<T>): Promise<T> => {
-        return this.repository
-            .findOne({
-                where: crudRecoverRequest.params as unknown as FindOptionsWhere<T>,
-                withDeleted: true,
-            })
-            .then(async (entity: T | null) => {
-                if (!entity) {
-                    throw new NotFoundException();
-                }
-                await this.repository.recover(entity);
-                return entity;
-            });
+        return this.findOne(crudRecoverRequest.params as unknown as FindOptionsWhere<T>, true).then(async (entity: T | null) => {
+            if (!entity) {
+                throw new NotFoundException();
+            }
+            await this.repository.recover(entity);
+            return entity;
+        });
     };
+
+    private async findOne(where: FindOptionsWhere<T>, withDeleted: boolean): Promise<T | null> {
+        if (!this.usableQueryRunner) {
+            return this.repository.findOne({ where, withDeleted });
+        }
+        const queryBuilder = this.repository.createQueryBuilder().where(where);
+        if (withDeleted) {
+            queryBuilder.withDeleted();
+        }
+        const runner = queryBuilder.connection.createQueryRunner('master');
```

Comment:
I think this new findOne method will make many operation(reservedRecover, reservedUpdate..etc) always use master's connection whenever the database type supports replication. 
Will there be no need for findOne to query using slave database?

Reply:
The goal of this PR to query using master database

Reply:
It seems like there is a contradiction between Goal you have written on PR description and the comment above ☝🏼.

> ## What is the new behavior?
> 
> If the DBMS supports replication,
> it avoids problem by findOne from the master node

Based on your work, I guess PR description should be fixed as 'it avoids problem by findOne from the ~~master~~ slave node'.

Reply:
oh, right!. thanks

Reply:
it means to findOne from master as a way to avoid the problem.
The comment seems vague to me too. 

thanks :)